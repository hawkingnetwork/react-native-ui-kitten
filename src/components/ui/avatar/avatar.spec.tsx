import React from 'react';
import {
  Image,
  ImageSourcePropType,
  StyleSheet,
} from 'react-native';
import {
  render,
  RenderAPI,
  shallow,
} from 'react-native-testing-library';
import { ReactTestInstance } from 'react-test-renderer';
import {
  ApplicationProvider,
  ApplicationProviderProps,
} from '@kitten/theme';
import {
  Avatar,
  AvatarProps,
} from './avatar.component';
import {
  mapping,
  theme,
} from '../support/tests';

const Mock = (props?: AvatarProps): React.ReactElement<ApplicationProviderProps> => {
  return (
    <ApplicationProvider
      mapping={mapping}
      theme={theme}>
      <Avatar {...props} />
    </ApplicationProvider>
  );
};

const renderComponent = (props?: AvatarProps): RenderAPI => {
  return render(
    <Mock {...props} />,
  );
};

const iconSource: ImageSourcePropType = { uri: 'https://akveo.github.io/eva-icons/fill/png/128/star.png' };

describe('@avatar: matches snapshot', () => {

  describe('* appearance', () => {

    it('* default', () => {
      const component: RenderAPI = renderComponent({
        source: iconSource,
      });

      const { output } = shallow(component.getByType(Avatar));

      expect(output).toMatchSnapshot();
    });

  });

});

describe('@avatar: component checks', () => {

  it('* round shape styled properly', () => {
    const component: RenderAPI = renderComponent({
      source: iconSource,
      shape: 'round',
    });

    const avatar: ReactTestInstance = component.getByType(Image);

    const { borderRadius, height } = StyleSheet.flatten(avatar.props.style);

    expect(borderRadius).toEqual(height / 2);
  });

  it('* rounded shape styled properly', () => {
    const component: RenderAPI = renderComponent({
      source: iconSource,
      shape: 'rounded',
    });

    const avatar: ReactTestInstance = component.getByType(Image);

    const { borderRadius, height } = StyleSheet.flatten(avatar.props.style);

    expect(borderRadius).toBeLessThan(height);
  });

  it('* square shape', () => {
    const component: RenderAPI = renderComponent({
      source: iconSource,
      shape: 'square',
    });

    const avatar: ReactTestInstance = component.getByType(Image);

    const { borderRadius } = StyleSheet.flatten(avatar.props.style);

    expect(borderRadius).toEqual(0);
  });

});
