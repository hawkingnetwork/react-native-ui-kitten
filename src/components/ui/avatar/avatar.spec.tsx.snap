// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@avatar: matches snapshot * appearance * default 1`] = `
<Wrapper
  forwardedRef={null}
  source={
    Object {
      "uri": "https://akveo.github.io/eva-icons/fill/png/128/star.png",
    }
  }
/>
`;
